### PR TITLE
i3bar-river: init at 0.1.3

### PR DESCRIPTION
**Pull request by nicegamer7**
## Description of changes
A port of i3bar for river, though I'm pretty sure it works on most Wayland compositors.
https://github.com/MaxVerevkin/i3bar-river

## Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -32978,6 +32978,8 @@ with pkgs;
 
   multilockscreen = callPackage ../misc/screensavers/multilockscreen { };
 
+  i3bar-river = callPackage ../applications/misc/i3bar-river { };
```
Comment (nagy) [inline pkgs/top-level/all-packages.nix:32981]: can you use the new by-name structure?

https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#name-based-package-directories

**Review by nagy — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (nicegamer7) [reply pkgs/top-level/all-packages.nix]: Done, could you take another look?

**Review by nagy — APPROVED** [review]

File: pkgs/by-name/i3/i3bar-river/package.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+, pango
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "i3bar-river";
+  version = "0.1.3";
+
+  src = fetchFromGitHub {
+    owner = "MaxVerevkin";
+    repo = pname;
```
Comment (wegank) [inline pkgs/by-name/i3/i3bar-river/package.nix:15]:
```suggestion
    repo = "i3bar-river";
```

File: pkgs/by-name/i3/i3bar-river/package.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ glib pango ];
+
+  meta = with lib; {
+    description = "A port of i3bar for river";
+    homepage = "https://github.com/MaxVerevkin/i3bar-river";
+    license = licenses.gpl3Only;
+    mainProgram = "i3bar-river";
+    platforms = platforms.unix;
```
Comment (wegank) [inline pkgs/by-name/i3/i3bar-river/package.nix:30]:
```suggestion
    platforms = platforms.linux;
```

Does not build on darwin, hasn't tested BSD.

File: pkgs/by-name/i3/i3bar-river/package.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  };
```
Comment (wegank) [inline pkgs/by-name/i3/i3bar-river/package.nix:31]: No maintainer?

**Review by wegank — APPROVED** [review]

**Pull request merged**